docs/monitoring.md

```diff
@@ -256,6 +256,157 @@ still required, though there is only one application available.  Eg. to see the
 running app, you would go to `http://localhost:4040/api/v1/applications/[app-id]/jobs`.  This is to
 keep the paths consistent in both modes.
 
+## Hadoop YARN Timeline service history provider
+
+As well as the Filesystem History Provider, Spark can integrate with the Hadoop YARN 
+"Application Timeline Service". This is a service which runs in a YARN cluster, recording
+application- and YARN- published events to a database, retrieving them on request.
+
+Spark integrates with the timeline service by
+1. Publishing events to the timeline service as applications execute.
+2. Listing application histories published to the timeline service.
+3. Retrieving the details of specific application histories.
+
+### Configuring the Timeline Service
+
+For details on configuring and starting the timeline service, consult the Hadoop documentation.
+
+From the perspective of Spark, the key requirements are
+1. The YARN timeline service must be running.
+1. Its URL is known, and configured in the `yarn-site.xml` configuration file.
```

Comment:
Markdown takes either; here's [the rendered file|https://github.com/steveloughran/spark/blob/stevel/feature/SPARK-1537-ATS/docs/monitoring.md]